web-app/js/portal/prototypes/Ext.js

```diff
@@ -34,5 +34,16 @@ Ext.tree.TreeNode.prototype.clone = function(recursive) {
     }
     
     return clone;
-}
+};
 
+Ext.tree.TreeNode.prototype.getHierarchy = function(attribute) {
```

Comment:
Yeah, it's an interesting one. Because the behaviour is generic (so it could be meaningful somewhere else), and because things get messy if you did want to isolate it to a subclass, I'm not worried about adding this new behaviour to TreeNode.

:+1: